borg/archive.py

```diff
@@ -403,26 +408,29 @@ def restore_attrs(self, path, item, symlink=False, fd=None):
         uid = item[b'uid'] if uid is None else uid
         gid = item[b'gid'] if gid is None else gid
         # This code is a bit of a mess due to os specific differences
-        try:
+        if sys.platform != 'win32':
+            try:
+                if fd:
+                    os.fchown(fd, uid, gid)
+                else:
+                    os.lchown(path, uid, gid)
+            except OSError:
+                pass
             if fd:
-                os.fchown(fd, uid, gid)
-            else:
-                os.lchown(path, uid, gid)
-        except OSError:
-            pass
-        if fd:
-            os.fchmod(fd, item[b'mode'])
-        elif not symlink:
-            os.chmod(path, item[b'mode'])
-        elif has_lchmod:  # Not available on Linux
-            os.lchmod(path, item[b'mode'])
+                os.fchmod(fd, item[b'mode'])
+            elif not symlink:
+                os.chmod(path, item[b'mode'])
+            elif has_lchmod:  # Not available on Linux
+                os.lchmod(path, item[b'mode'])
         mtime = bigint_to_int(item[b'mtime'])
         if b'atime' in item:
             atime = bigint_to_int(item[b'atime'])
         else:
             # old archives only had mtime in item metadata
             atime = mtime
-        if fd:
+        if sys.platform == 'win32':
+            os.utime(path, ns=(atime, mtime))
```

Comment:
this is the same call as in the else branch.

the 2nd argument (None) given there defaults to None.

https://docs.python.org/3.4/library/os.html